LinqToRest/ResourceRetriever.cs

```diff
@@ -106,9 +108,42 @@ private static Uri GetResourceUri(JToken token, Uri root)
         private object GetDeserializedParameter(ParameterInfo parameter, JToken token, Uri uri)
         {
             var type = parameter.ParameterType;
+
             return type.IsQueryable()
                 ? _queryableFactory(type.GetInnerType(), uri)
-                : GetField(type, token, parameter.Name);
+                : type.IsEnum
+                    ? GetEnumFromType(type, token, parameter.Name)
+                    : GetField(type, token, parameter.Name);
+        }
+
+        private static object GetEnumFromType(Type type, JToken token, string name)
+        {
+            var getEnum = typeof(ResourceRetriever).GetMethod(nameof(GetEnum))?.MakeGenericMethod(type);

```

Comment:
Could you move the `?? throw new InvalidOperationException()` right after the `GetMethod(...)` call? That way, we don't propagate any `null` values and don't need to `throw` inside the `try` statement.

Reply:
Adressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/15/commits/e32cccdcc20fcceb77141a33f1d8c11c502c9661